python/ray/serve/tests/test_cli_2.py

```diff
@@ -964,5 +964,27 @@ def test_serve_run_mount_to_correct_deployment_route_prefix(ray_start_stop):
     )
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+def test_control_c_shutdown_serve_components(ray_start_stop):
```

Comment:
I presume you tested that this fails w/o your change?

Reply:
yep, I purposely push a commit with just the test trying to see this failing in CI with the serve component still alive as well https://buildkite.com/ray-project/premerge/builds/25316#018f3652-b00a-45a2-aadc-ca46fb8d4f5d/182-3706